CMakeLists.txt

```diff
@@ -49,12 +49,25 @@ set(SRC_CPPCLI
   "djinni/cppcli/WrapperCache.cpp"
 )
 
-option(DJINNI_STATIC_LIB "Build Djinni support library as a static library instead of dynamic (the default)." off)
+# The target type of the library can be defined by setting `DJINNI_STATIC_LIB`.
+# If `DJINNI_STATIC_LIB` is not defined, the default CMake flag `BUILD_SHARED_LIBS` is followed.
+if(NOT DEFINED DJINNI_STATIC_LIB)
+  if(BUILD_SHARED_LIBS)
+    set(DJINNI_STATIC_LIB OFF)
+  else()
+    set(DJINNI_STATIC_LIB ON)
+  endif()
+endif()
+
```

Comment:
ok, I decided to go down another path than what we discussed in slack to resolve #23. It was a priority for me to not break existing setups. And because the variable `DJINNI_STATIC_LIB` is used twice in the configuration of the target, it has to be set anyways. That's why it seemed the easiest solution to just set it according to `BUILD_SHARED_LIB` if undefined. This does not use the default CMake mechanism when defining the target, but simulates it.

What do you think about that? I have mixed feelings. The downside of this is that it introduces additional complexity. Reducing complexity would only be possible by introducing a breaking change. I think I have at least figured out a solution that introduces as little additional code as possible.

Reply:
Btw do you know why the ObjC flavor of the lib is only linked to the Foundation framework if built as shared lib? 
```
  if (NOT DJINNI_STATIC_LIB)
    target_link_libraries(djinni_support_lib
      "-framework Foundation"
    )
  endif()
```

The only explanation that I have is: *"If this is a static lib the library that will link to this will most likely link the Foundation framework anyways. So screw it here."*

Reply:
I think what you suggested on slack, second way, is the best solution.
If it breaks things, lets break it (with a new version number). 
Complexity kills, especially if it is not required, and this is not required since it gives us no great functionality.
For a project like that , where people drop out and in, and you might have a breaks between reading the code where you lose the context, readability and maintainability must be the number 1 priority.

--
If you have a static lib and use it, yourself needs to link against foundation yourself
Your app will only take from the static lib what it needs, and embed it. 

If you use a dynamic library nothing gets embedded into your app, it just says I need these symbols and expect them to be found at runtime. The dynamic library does the same with an other dynamic lib it depends on.

But a static library does/can not do that, it says, if you use me, I expect you to provide the external symbol, either via an other static library at link time (than pls get the link order right) or at runtime via an other dynamic library